upload-api/functions/ucan-invocation-router.js

```diff
@@ -322,21 +323,32 @@ export async function ucanInvocationRouter(request) {
   const storageProviderTable = createStorageProviderTable(AWS_REGION, storageProviderTableName, options)
   const routingService = createRoutingService(storageProviderTable, serviceSigner)
 
-  /** @type {Array<import('@storacha/upload-api/types').SSOProvider & { name: string }>} */
+  /** @type {Array<import('@storacha/upload-api/types').SSOProvider>} */
   const ssoProviders = []
   if (process.env.DMAIL_API_KEY && process.env.DMAIL_API_SECRET) {
+
     const dmailSSOService = createDmailSSOService({
-      DMAIL_API_KEY: mustGetEnv('DMAIL_API_KEY'),
-      DMAIL_API_SECRET: mustGetEnv('DMAIL_API_SECRET'),
-      DMAIL_JWT_SECRET: process.env.DMAIL_JWT_SECRET || 'unused', // if undefined, we set it to a dummy value to bypass JWT validation
-      DMAIL_API_URL: process.env.DMAIL_API_URL || 'https://api.dmail.ai/open/api/storacha/getUserStatus',
+      apiKey: Config.DMAIL_API_KEY,
+      apiSecret: Config.DMAIL_API_SECRET,
+      jwtSecret: Config.DMAIL_JWT_SECRET || 'unused', // if undefined, we set it to a dummy value to bypass JWT validation
+      apiUrl: process.env.DMAIL_API_URL || 'https://api.dmail.ai/open/api/storacha/getUserStatus',
     })
     ssoProviders.push(dmailSSOService)
   }
+
+  // TODO (fforbeck): if more providers are added, we need to add a feature flag for each provider using a list of providers names
+  // then if the provider is in the list, we enable the customer trial plan for that provider
+  const enableCustomerTrialPlan = process.env.ENABLE_CUSTOMER_TRIAL_PLAN === 'true'
+
+  // Build service URL from request data since we're in the same ucan server
+  const requestHost = request.headers?.host || request.headers?.Host
+  const serviceUrl = requestHost ? new URL(`https://${requestHost}`) : uploadServiceURL
```

Comment:
You can use the already defined `accessServiceURL` for this which is set the whichever URL was requested in the invocation.

https://github.com/storacha/w3infra/blob/847a6b24ba023396ea3106cc743925e75b10a095/stacks/upload-api-stack.js#L129

Reply:
sounds good. I'll use that. thanks